examples/smartContracts/deployer.ts

```diff
@@ -24,85 +57,75 @@ export const deploySmartContract = async (
 			deployerAccount = web3Client.wallet().getBaseAccount();
 		}
 
-		// set default values in case of missing ones
-		contractData.maxGas = contractData.maxGas || 200000;
-		contractData.fee = contractData.fee || 0;
+		// check deployer account balance
+		await checkBalance(web3Client, deployerAccount);
 
-		// construct a sc utils helper
-		const utils = new SmartContractUtils();
+		// construct a new datastore
+		const datastore = new Map<Uint8Array, Uint8Array>();
 
-		// compile deployement sc from wasm
-		console.log(`Running ${chalk.green("compilation")} of smart contract on path ${chalk.yellow(deploymentScWasm)}....`);
-		let compiledSc: ICompiledSmartContract;
-		try {
-			compiledSc = await utils.compileSmartContractFromWasmFile(deploymentScWasm);
-			console.log(`Smart Contract ${chalk.green("successfully")} compiled to deployable bytecode under ${deploymentScWasm}`);
-		} catch (ex) {
-			const msg = chalk.red(`Error compiling smart contract on path ${chalk.yellow(deploymentScWasm)}`);
-			console.error(msg);
-			throw new Error(ex);
-		}
-		if (!compiledSc.binary) {
-			const msg = chalk.red(`No bytecode to deploy for wasm file ${chalk.yellow(deploymentScWasm)}. Check AS compiler`);
-			console.error(msg);
-			throw new Error(msg);
-		}
+		// set the number of contracts
+		datastore.set(
+		  new Uint8Array([0x00]),
+		  new Uint8Array(new Args().addU64(BigInt(contractsToDeploy.length)).serialize()),
+		);
+		// loop through all contracts and fill datastore
+		for (let i = 0; i < contractsToDeploy.length; i++) {
+			const contract: ISCData = contractsToDeploy[i];
 
-		// compile deployed sc from wasm
-		console.log(`Running ${chalk.green("compilation")} of smart contract on path ${chalk.yellow(deployedScWasm)}....`);
-		let compiledDeployedSc: ICompiledSmartContract;
-		try {
-			compiledDeployedSc = await utils.compileSmartContractFromWasmFile(deployedScWasm);
-			console.log(`Smart Contract ${chalk.green("successfully")} compiled to deployable bytecode under ${deployedScWasm}`);
-		} catch (ex) {
-			const msg = chalk.red(`Error compiling smart contract on path ${chalk.yellow(deployedScWasm)}`);
-			console.error(msg);
-			throw new Error(ex);
-		}
-		if (!compiledDeployedSc.binary) {
-			const msg = chalk.red(`No bytecode to deploy for wasm file ${chalk.yellow(deployedScWasm)}. Check AS compiler`);
-			console.error(msg);
-			throw new Error(msg);
+			datastore.set(
+				new Uint8Array(new Args().addU64(BigInt(i + 1)).serialize()),
+				contract.data,
+			);
+			if (contract.args) {
+				datastore.set(
+					new Uint8Array(
+					new Args()
+						.addU64(BigInt(i + 1))
+						.addUint8Array(new Uint8Array([0x00]))
+						.serialize(),
+					),
```

Comment:
new Uint8Array([...u64ToBytes(BigInt(i + 1)), 0])

Reply:
discussed and fixed